src/weatherDetails/DetailsPanel.tsx

```diff
@@ -0,0 +1,60 @@
+import { Footer } from "./Footer";
+import { UnitButton } from "./UnitButtons";
+import { Forecasts } from "./Forecasts";
+import { CurrentWeatherHighlights } from "./CurrentWeatherHighlights";
+import { ErrorAlert } from "./ErrorAlert";
+import "./DetailsPanel.css";
+import { CurrentWeather, ForecastWeather } from "../App";
+import { useMemo } from "react";
+
+type DetailsPanelProps = {
+  fetchError: Error | null;
+  currentWeather: CurrentWeather | null;
+  forecastWeather: ForecastWeather | null;
+  units: string;
+  onSetUnits: React.Dispatch<React.SetStateAction<string>>;
+};
+
+export const DetailsPanel = ({
+  fetchError,
+  onSetUnits,
+  forecastWeather,
+  units,
+  currentWeather,
+}: DetailsPanelProps) => {
+  const activeUnit = useMemo(() => {
+    if (units === "metric") return "C";
```

Comment:
What if I make a typo and write `units === 'metrics` ? I won't have an error, but I should have an error